skimage/draw/_draw.pyx

```diff
@@ -398,18 +340,20 @@ def _circle_perimeter_aa(Py_ssize_t r_o, Py_ssize_t c_o,
         val.extend([1 - dceil, dceil] * 8)
```

Comment:
Is there a better way to get unique coordinates than to check the whole array for uniqueness?

Reply:
Maybe. I don't fully understand the algorithms that generate the duplicates. It's easy to see why some duplicates are generated (e.g. in `_circle_perimeter` if one of the variables `r` or `c` is 0 the leading signs become meaningless).

Do you feel that performance is a problem here? I did a quick test and using `np.unique` slows down `circle_perimeter` by the factor 4 (performance seems similar when commenting that out). I'm surprised that `np.unique` has that big of an impact.

Reply:
i guess you could check if r == c i many cases, but in the aa case it is a little trickier.

Reply:
That's an idea. But we have to check for 0 as well. This

```python
if r == c == 0:
    rr.append(r)
    cc.append(c)
elif r == 0:
    rr.extend([r, r, c, -c])
    cc.extend([c, -c, r, r])
elif c == 0:
    rr.extend([r, -r, c, c])
    cc.extend([c, c, r, -r])
elif r == c:
    rr.extend([r, -r, r, -r])
    cc.extend([c, c, -c, -c])
else:
    rr.extend([r, -r, r, -r, c, -c, c, -c])
    cc.extend([c, c, -c, -c, r, r, -r, -r])
```

seems to work. Did a few quick tests and this is nearly the same as the old implementation. The `np.unique` approach is consistently 3 to 4x slower.

Reply:
That solution would work for `circle_perimeter_aa` as well but the cases would get kind of tedious because we'd have to consider `r - 1` and `1 - r` as well...